src/stores/NotificationsStore.js

```diff
@@ -0,0 +1,280 @@
+      }),
+      headers: {
+        Accept: 'application/json',
+        Authorization: `Bearer ${AuthStore.token || ''}`,
+        'Content-Type': 'application/json',
+      },
+      method: 'post',
+    });
+  };
+
+  unregisterToken = () =>
+    fetchJSON({
+      method: 'post',
+      url: `${BASE_PUSH_URL}/unregister`,
+    });
+
+  onNotificationPress = async (notification: Notification): Promise<void> => {
+    await this._waitForInitializationFinish();
+    this._handleNotificationPressByType(notification);
+  };
+
+  @action
+  setNavigation = (navigation: Navigation) => {
+    this._navigation = navigation;
+  };
+
+  @action deleteAll = () => this._notificationsByID.clear();
+
+  @action deleteByID = (id: string) => this._notificationsByID.delete(id);
+
+  @action
+  setRead = (notificationID: string) => {
+    const notification = this._notificationsByID.get(notificationID);
+    if (!notification) {
+      return;
+    }
+
+    this._notificationsByID.set(notificationID, {
+      ...notification,
+      isRead: true,
+    });
+  };
+
+  @action
+  _addNotification = (notification: Notification): void =>
+    this._notificationsByID.set(notification.id, notification);
+
+  _rehydrateNotifications = async () => {
+    const notificationsString = await AsyncStorage.getItem(
+      NOTIFICATIONS_STORAGE_KEY,
+    );
+
+    const notifications = notificationsString
+      ? JSON.parse(notificationsString).map(notification => [
+          notification.id,
+          notification,
+        ])
+      : [];
+
+    runInAction(() => {
+      this._notificationsByID.merge((notifications: any));
+    });
+  };
+
+  _waitForInitializationFinish = (): Promise<void> =>
+    new Promise((resolve: () => void) => {
+      if (this._isInitialized) {
+        resolve();
+        return;
+      }
+      when(() => this._isInitialized, () => resolve());
+    });
+
+  _onRawNotification = rawNotification => {
+    // ignore empty callbackNotification call when open the app
+    // from main icon when the app is in background currently
+    if (rawNotification.fcm.action === 'android.intent.action.MAIN') {
+      return;
+    }
+
+    const parsedNotification = rawNotification.custom_notification
+      ? JSON.parse(rawNotification.custom_notification)
+      : rawNotification;
+
+    const existingNotification = this._notificationsByID.get(
+      parsedNotification.id,
+    );
+
+    const openedFromTray = !!parsedNotification.opened_from_tray;
+
+    this._addNotification({
+      ...parsedNotification,
+      date: existingNotification ? existingNotification.date : new Date(),
+      isRead: openedFromTray,
+    });
+
+    if (openedFromTray) {
+      this.onNotificationPress(parsedNotification);
+    }
+  };
+
+  _handleNotificationPressByType = (notification: Notification) => {
+    switch (notification.type) {
+      case 'lowKegLevel': {
+        const { tapId }: LowKegLevelNotification = (notification: any);
+        KegStore.flushCache();
```

Comment:
Shouldn't you just flush the keg for a specific tapId? I could even send the kegID in the push notification.

Reply:
yeah. it should be more precise, but we don't have flushEntityCacheByID() method in DAO right now.